src/Mail/SmtpMailer.php

```diff
@@ -145,9 +146,19 @@ protected function connect()
 		}
 
 		if ($this->username != NULL && $this->password != NULL) {
-			$this->write('AUTH LOGIN', 334);
-			$this->write(base64_encode($this->username), 334, 'username');
-			$this->write(base64_encode($this->password), 235, 'password');
+			$authMechanisms = [];
+			if (preg_match('~^250[ -]AUTH (.*)$~im', $ehloResponse, $matches)) {
+				$authMechanisms = explode(' ', trim($matches[1]));
+			}
+
+			if (array_search('PLAIN', $authMechanisms, true) !== false) {
```

Comment:
Alright, I will change it to in_array.

I also though about it and I don't think that there is a reason to prefer LOGIN mechanism over PLAIN. They both offer same security. The PLAIN mechanism will be just a bit faster ... correct me if I'm wrong.